modules/images/regenerate-existing-images/background-process/class-perflab-background-job.php

```diff
@@ -0,0 +1,314 @@
+<?php
+/**
+ * Background job class.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * Class Perflab_Background_Job.
+ *
+ * Manage and run the background jobs.
+ *
+ * Following fields are being stored for a background job:
+ *
+ * 1. Job ID: Identifies the job; stored as the term ID.
+ * 2. Job action: This is the custom action string that is called in the background process. This is prefixed so the final action hooked called is `do_action( 'perflab_job_{$job_action}' )`.
```

Comment:
@felixarntz Yes, I think adding job_action as term_name will be good approach for faster lookup.
However; for slug, I think better approach would be to use `wp_unique_term_slug` as it will be suffixed by a number like `job-action-2`, `job-action-3` etc for similar name actions.

CC: @jjgrainger

Reply:
@felixarntz @ankitrox I agree and this will help simplify our approach as we can remove the Job Name from the API.

I like the idea of using `wp_unique_term_slug` to generate a unique slug for the job, rather than creating our own.

Reply:
@ankitrox @jjgrainger Makes sense. I suggest let's implement the storage like https://github.com/WordPress/performance/pull/507#discussion_r983620362.